linux_new/Jenkinsfile

```diff
@@ -0,0 +1,959 @@
+                    // Tar And Publish The Generated Build Files
+                    // May Need Rework
+                       sh "tar -czf ./package_build_files.tar.gz ./linux_new/*"
+                       sh "pwd"
+                       sh "ls -ltr"
+                       // Publish the tarball
+                       archiveArtifacts artifacts: "package_build_files.tar.gz", allowEmptyArchive: false
+                    // Stash The Generated Build Files
+                    dir('linux_new') {
+                    stash name: 'installercode', includes: '**'
+                    }
+                  }
+              }
+          }
+// Generate Spec File Stage - End
+// Build And Archive Packages Stage - Start
+stage('Build & Archive Package') {
+    // when {
+    //     expression { return params.RELEASE }
+    // }
+    steps {
+        script {
+            DISTS_TO_BUILD.each { DistArrayElement ->
+                def nodeLabel = ''
+                if (DistArrayElement == 'alpine') {
+                    nodeLabel = PKGBUILDLABELAPK
+                } else if (DistArrayElement == 'debian') {
+                    nodeLabel = PKGBUILDLABELDEB
+                } else if (DistArrayElement in ['rhel', 'suse']) {
+                    nodeLabel = PKGBUILDLABELRHEL
+                } else {
+                    error "Unknown DistArrayElement: ${DistArrayElement}"
+                }
+
+                node(label: nodeLabel) { // Assign node dynamically based on distribution type
+                    // echo "Build Packages For Arch: ${arch}"
+                    // echo "Build Packages For PARCH: ${packagearch}"
+                    // echo "Building For Dist: ${DistArrayElement}"
+                    // echo "Build Product: ${PRODUCT}"
+                    // echo "Build Version : ${Release}"
+
+                    // Docker --mount option requires BuildKit
+                    env.DOCKER_BUILDKIT = 1
+                    env.COMPOSE_DOCKER_CLI_BUILD = 1
+
+                    // Prepare Workspace
+                    cleanWs()
+                    unstash 'installercode'
+
+                    try {
+                        def PackagesToBuild = ['jdk', 'jre']
+                        def buildCli
+                        PackagesToBuild.each { PackageArrayElement ->
+                            echo "Building Package: ${PackageArrayElement} for ${DistArrayElement}"
+                            def gBuildTask = (packagearch in ['x86_64', 'amd64']) ?
+                                "package${PackageArrayElement}${DistArrayElement} check${PackageArrayElement}${DistArrayElement}" :
+                                "package${PackageArrayElement}${DistArrayElement}"
+                            
+                            // Override Package Arch To Be Dist Specific
+
+                            if (DistArrayElement == "rhel" || DistArrayElement == "suse") {
+                              echo "Using build CLI for RHEL/SUSE: ${buildCli}"
```

Comment:
Maybe a stupid question but should this be displaying `buildCli` just before it's replaced? (Same comment for other occurrences of the same pattern)

Reply:
They were there just for development, I want to check the commands prior to having jenkins execute them, so I've removed them now Im happy with the way they work.